contrib/win32/win32compat/misc.c

```diff
@@ -1466,3 +1466,77 @@ is_absolute_path(char *path)
 	
 	return retVal;
 }
+
+/* return 0 - in case of failure */
+BOOL
+create_directory_withsddl(char *path, char *sddl)
+{
+	struct stat st;
+	if (stat(path, &st) < 0) {
```

Comment:
from earlier comment... stat() can return -1 when OOM. 
Instead of relying on this call, you directly call CreateDirectory and ignore its return value.

Reply:
CreateDirectory() return value is very important as the next steps like copy of sshd_config_default, etc are dependent on this and its a cleaner approach to look for its existence.

since OOM is very rare and even if we encounter OOM then these functions will succeed but it will fail down the line, so why not use stat?

Reply:
As we discussed, I am ignoring this comment.
FYI, stat() is used like this everywhere in the openssh code..